app/presenters/openid_connect_user_info_presenter.rb

```diff
@@ -8,17 +8,19 @@ def initialize(identity)
   end
 
   def user_info
+    scoper = OpenidConnectAttributeScoper.new(identity.scope)
     info = {
       sub: uuid_from_sp_identity(identity),
       iss: root_url,
       email: email_from_sp_identity(identity),
       email_verified: true,
-      verified_at: verified_at,
-    }.
-           merge(x509_attributes).
-           merge(ial2_attributes)
+    }
+
+    info.merge!(ial2_attributes) if scoper.ial2_scopes_requested?
+    info.merge!(x509_attributes) if scoper.x509_scopes_requested?
+    info[:verified_at] = verified_at if scoper.verified_at_requested?
```

Comment:
I think this couples these objects in a weird way...here's an idea that uses procs for lazy evaluation so we could move this loading inside?

```ruby
OpenidConnectAttributeScoper.new(identity.scope).filter(
  info: info,
  ial2_attributes_loader: proc { ial2_attributes },
  x509_attributes_loader: proc { x509_attributes },
  verified_at_loader: proc { verified_at }
)
```

and the inside of `.filter` we would do like `ial2_attributes_loader.call if ial2_scopes_requested?` kinda thing?

Reply:
It's a weird interaction, and I struggled a bit with how to optionally handle it. I think the separation of data loading is important though, and I'm not sure I'd want to include the data mutation or execution of database queries in the scoper class.

I'm open to changes, but I'm not sure that the classes are any more coupled than they were before. The scoper parses/validates the `scope` attribute, and ultimately transforms the hash received based on the `scope`.

I'm not entirely opposed to the idea of combining the two classes.

Reply:
Yeah maybe that's the solution, just combine the two objects, I feel like we use attribute scoping separately from presenting, so as long as we expose that method it should be fine?